### DISCUSSION TITLE
Generate and render  a tree dynamically

### DISCUSSION THREAD

**Discussion by NikosAlexandris** [pawamoy/markdown-exec#44] [Q&A] [answered]
Is possible in some way to dynamically generate the content for a 
````
```tree
```
````
code-block ?

All of the examples in https://pawamoy.github.io/markdown-exec/usage/tree/ are static ones.

I've written a small function that prints the content of a directory. Of course it can be further customised to match the expected `tree` code-block structure.

Example

```` python
```python exec="true"
from pathlib import Path


def print_tree(directory, level=1):
    directory = Path(directory)
    if not directory.is_dir():
        return  # Ensure it's a directory

    # Only list the first level of files and directories
    tree = str()
    for path in sorted(directory.iterdir()):
        if path.name.startswith('_'):
            continue  # Skip files or directories starting with '_'
        if path.is_dir():
            tree += f"\n    {path.name}/"  # Add slash to indicate it's a directory
        else:
            tree += f"\n    {path.name}"

    # Optionally print a trailing line after listing
    if level == 1:
        tree += '\n   '

    return tree

tree = print_tree('api/')
print(f"```\n{tree}\n```")
```
````

**Comment by pawamoy** [comment, marked as answer]
> I've written a small function that prints the content of a directory. Of course it can be further customised to match the expected tree code-block structure.

So, does it work :slightly_smiling_face:?

Instead of using `print(f"```\n{tree}\n```")` I think you can simply print the tree without fences, and add `result="tree"` to your code block options.

**Comment by NikosAlexandris** [comment]
Yes, it does

![2024-04-21T13:44:39,621353293+03:00](https://github.com/pawamoy/markdown-exec/assets/7046639/f369fded-292c-40e8-9f65-05ba4f5ea70c)

and your _library_ is Awesome.

ps- And I need to fix my function to work properly :D.

**Reply by pawamoy** [reply]
Great! And thanks :blush:
